test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/SharedServiceStateTests.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Sdk.TestFramework;
+using Moq;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+using Test.Utility;
+using Xunit;
+
+namespace NuGet.PackageManagement.VisualStudio.Test
+{
+    [Collection(MockedVS.Collection)]
+    public class SharedServiceStateTests : MockedVSCollectionTests
+    {
+        public SharedServiceStateTests(GlobalServiceProvider globalServiceProvider)
+            : base(globalServiceProvider)
+        {
+            var solutionManager = new Mock<IVsSolutionManager>();
+
+            solutionManager.SetupGet(x => x.SolutionDirectory)
```

Comment:
Does the current test suite fail without a SolutionDirectory setup?

Reply:
No, because existing tests don't use Moq but a test implementation of `IVsSolutionManager`.

https://github.com/NuGet/NuGet.Client/blob/34b6220ba23af326f079eb1628593de06234fd33/test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetProjectManagerServiceTests.cs#L672

Reply:
Ok, my suggestion would be to remove the Setup if it's not doing anything needed.